### PR TITLE
mvebu: add support for MACCHIATObin (cortex-a72)

### PR DESCRIPTION
**Pull request by damirsamardzic**
Add initial support for Marvell MACCHIATObin, cortex-a72 based Marvell
ARMADA 8040 Community board. Comes in two forms: Single Shot and Double
Shot.

Specifications:
- Quad core Cortex-A72 (up to 2GHz)
- DDR4 DIMM slot with optional ECC and single/dual chip select support
- Dual 10GbE (1/2.5/10GbE) via copper or SFP
  2.5GbE (1/2.5GbE) via SFP
  1GbE via copper
- SPI Flash
- 3 X SATA 3.0 connectors
- MicroSD connector
- eMMC
- PCI x4 3.0 slot
- USB 2.0 Headers (Internal)
- USB 3.0 connector
- Console port (UART) over microUSB connector
- 20-pin Connector for CPU JTAG debugger
- 2 X UART Headers
- 12V input via DC Jack
- ATX type power connector
- Form Factor: Mini-ITX (170 mm x 170 mm)

More details at http://macchiatobin.net

Booting from micro SD card:
 1. reset U-Boot environment:
      env default -a
      saveenv

 2. prepare U-Boot with boot script:
      setenv bootcmd "load mmc 1:1 0x4d00000 boot.scr; source 0x4d00000"
      saveenv

    or manually:
      setenv fdt_name armada-8040-mcbin.dtb
      setenv image_name Image
      setenv bootcmd 'mmc dev 1; ext4load mmc 1:1 $kernel_addr $image_name;ext4load mmc 1:1 $fdt_addr $fdt_name;setenv   bootargs $console root=/dev/mmcblk1p2 rw rootwait; booti $kernel_addr - $fdt_addr'
      saveenv

Signed-off-by: Damir Samardzic <damir.samardzic@sartura.hr>

### CODE REVIEW

File: target/linux/mvebu/cortexa72/config-default [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+# CONFIG_ARM64_PTDUMP_CORE is not set
+# CONFIG_ARM64_PTDUMP_DEBUGFS is not set
+# CONFIG_ARM64_RANDOMIZE_TEXT_OFFSET is not set
+# CONFIG_ARM64_SW_TTBR0_PAN is not set
+# CONFIG_ARM64_UAO is not set
+CONFIG_ARM64_VA_BITS=39
+CONFIG_ARM64_VA_BITS_39=y
+# CONFIG_ARM64_VA_BITS_48 is not set
+# CONFIG_ARM64_VHE is not set
+CONFIG_ARMADA_37XX_CLK=y
```
Comment (tmn505) [inline target/linux/mvebu/cortexa72/config-default]: You can drop this symbol,

File: target/linux/mvebu/cortexa72/config-default [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+CONFIG_ARMADA_AP806_SYSCON=y
+CONFIG_ARMADA_CP110_SYSCON=y
+CONFIG_ARM_AMBA=y
+CONFIG_ARM_ARCH_TIMER=y
+CONFIG_ARM_ARCH_TIMER_EVTSTREAM=y
+CONFIG_ARM_ARMADA_37XX_CPUFREQ=y
```
Comment (tmn505) [inline target/linux/mvebu/cortexa72/config-default]: this also,

File: target/linux/mvebu/cortexa72/config-default [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+CONFIG_NET_DSA_MV88E6XXX_GLOBAL2=y
+# CONFIG_NET_DSA_SMSC_LAN9303_I2C is not set
+CONFIG_NET_DSA_TAG_DSA=y
+CONFIG_NET_DSA_TAG_EDSA=y
+CONFIG_NET_SWITCHDEV=y
+# CONFIG_NUMA is not set
+CONFIG_PARTITION_PERCPU=y
+CONFIG_PCIEAER=y
+CONFIG_PCIE_ARMADA_8K=y
+CONFIG_PCI_AARDVARK=y
```
Comment (tmn505) [inline target/linux/mvebu/cortexa72/config-default]: this can be marked as not set,

File: target/linux/mvebu/cortexa72/config-default [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+CONFIG_PCI_BUS_ADDR_T_64BIT=y
+CONFIG_PGTABLE_LEVELS=3
+CONFIG_PHY_MVEBU_CP110_COMPHY=y
+CONFIG_PHYS_ADDR_T_64BIT=y
+CONFIG_PINCTRL_ARMADA_37XX=y
```
Comment (tmn505) [inline target/linux/mvebu/cortexa72/config-default]: this can be dropped,

File: target/linux/mvebu/cortexa72/config-default [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+CONFIG_POWER_SUPPLY=y
+# CONFIG_RANDOMIZE_BASE is not set
+CONFIG_REGULATOR_GPIO=y
+# CONFIG_SERIAL_AMBA_PL011 is not set
+CONFIG_SPARSEMEM=y
+CONFIG_SPARSEMEM_EXTREME=y
+CONFIG_SPARSEMEM_MANUAL=y
+CONFIG_SPARSEMEM_VMEMMAP=y
+CONFIG_SPARSEMEM_VMEMMAP_ENABLE=y
+CONFIG_SPI_ARMADA_3700=y
```
Comment (tmn505) [inline target/linux/mvebu/cortexa72/config-default]: and this one.
I'm wondering if `I2C_MUX_PCA954x` shouldn't be enabled => https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git/tree/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts?h=v4.14.30#n140.

File: target/linux/mvebu/image/cortex-a72.mk [hunk 6]
```diff
@@ -0,0 +1,16 @@
+ifeq ($(SUBTARGET),cortexa72)
+
+define Device/armada-macchiatobin
+  KERNEL_NAME := Image
+  KERNEL := kernel-bin
+  DEVICE_TITLE := MACCHIATObin (SolidRun Armada 8040 Community Board)
+  DEVICE_PACKAGES := e2fsprogs ethtool mkf2fs kmod-fs-vfat kmod-usb2 kmod-usb3 kmod-usb-storage kmod-mmc mmc-utils pciutils
```
Comment (tmn505) [inline target/linux/mvebu/image/cortex-a72.mk]: Remove packages from community repository `mmc-utils`, `pciutils`. The USB drivers are built in the kernel, should be safe to remove them.

**Review by tmn505 — CHANGES_REQUESTED** [review]
Otherwise looks good.
Are the SFP ports working?

**Comment by damirsamardzic** [comment]
Thanks for the review, I have tested with proposed changes, looks OK.
SFP ports don't work in this PR.

**Comment by hauke** [comment]
Please squash this together into one commit

**Review by tmn505 — APPROVED** [review]
Now it's all good.

**Comment by hauke** [comment]
Thank you for the patches, it was applied to master.

**Pull request closed without merging**